src/main/ruby/core/exception.rb

```diff
@@ -390,37 +401,56 @@ class KeyError < IndexError
 
 class SignalException < Exception
 
+  alias :signm :message
   attr_reader :signo
-  attr_reader :signm
-
-  def initialize(signo = nil, signm = nil)
-    # MRI overrides this behavior just for SignalException itself
-    # but not for anything that inherits from it, therefore we
-    # need this ugly check to make sure it works as intended.
-    return super(signo) unless self.class == SignalException
-    if signo.is_a? Integer
-      unless @signm = Signal::Numbers[signo]
-        raise ArgumentError, "invalid signal number #{signo}"
+
+  def initialize(*args)
+    argnum = 1
+    sig = nil
+    signo = 0
+    signm = nil
+    if args.size > 0
+      sig = Rubinius::Type.rb_check_to_integer(args.first, :to_int)
+      if sig.nil?
+        sig = args[0]
+      else
+        argnum = 2
       end
-      @signo = signo
-      @signm = signm || "SIG#{@signm}"
-    elsif signo
-      if signm
-        raise ArgumentError, 'wrong number of arguments (2 for 1)'
+    end
+    Rubinius::Type.check_arity(args.size, 1, argnum)
+    if argnum == 2
+      signo = Rubinius::Type.rb_num2int(sig)
+      if signo < 0 || signo > (Signal::Names.values.max + 1)
```

Comment:
The max Signal value should be computed only once.

Reply:
Isn't it enough to just check if `signo` is in `Signal::Numbers`?

Reply:
Fixed, to use `Signal::Numbers`